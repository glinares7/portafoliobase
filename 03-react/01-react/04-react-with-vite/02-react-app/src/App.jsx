// import { useState } from 'react'

//* fragment -component
// import { useState, Fragment } from "react";

import { useState } from "react";

import logo from "./logo.svg";
import "./App.css";

//* v1

// function App() {
//   // const [count, setCount] = useState(0)

//   return (
//     <div className="App">
//       <header className="App-header">
//         <img src={logo} className="App-logo" alt="logo" />
//         <pre>{new Date().getHours()}</pre>
//       </header>
//     </div>
//   );
// }

//* estilos de la aplicacion
const App = () => {
  //* destructuring of array -estado (cambio)
  const [contador, setContador] = useState(50);

  const disminuir = () => {
    setContador(contador - 1);
    // console.log(estado);
  };
  const aumentar = () => {
    setContador(contador + 1);
    // console.log(estado);
  };
  //* fragment vacio  (<> </>)
  return (
    <div className="App">
      <header className="App-header">
        <img src={logo} width="250" className="App-logo" alt="logo" />
        <div>
          <button onClick={disminuir}> - </button>
          <h1>{contador}</h1>
          <button onClick={aumentar}> + </button>
        </div>
      </header>
    </div>
  );
};

export default App;
