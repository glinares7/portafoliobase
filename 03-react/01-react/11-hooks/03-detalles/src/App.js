//* FUNCTIONAL COMPONENTS
//TODO rafce

// import Custom from "./components/menos/Custom";
import Memorize from "./components/menos/Memorize";

// import Ref from "./components/Ref";

// import LayoutEffect from "./components/LayoutEffect";

// import React from "react";
// import Effect from "./components/Effect";

// import { State } from "./components/State";
// import Calculadora from "./components/Calculadora";

//* COMPONENTES COMENTADOS
// <Effect />
// <LayoutEffect />;
// <Ref />
// <Custom />
const App = () => {
  return (
    <div className="container text-center">
      <Memorize />
    </div>
    // <div className="container text-center">
    //   <h1>Calculadora - PWA</h1>
    //   <hr />
    //   <Calculadora />
    // </div>

    // <State />
  );
};

export default App;
