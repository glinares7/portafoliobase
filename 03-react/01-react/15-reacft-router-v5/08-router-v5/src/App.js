import React, { useEffect, useReducer } from "react";
import LoginRouter from "./routes/LoginRouter";
import { AuthContext } from "./context/AuthContext";
import { authReducer } from "./reducers/AuthReducer";

const init = () => {
  return JSON.parse(localStorage.getItem("log")) || { log: false };
};
const App = () => {
  const [log, dispatch] = useReducer(authReducer, {}, init);
  useEffect(() => {
    localStorage.setItem("log", JSON.stringify(log));
  }, [log]);

  return (
    <AuthContext.Provider value={{ log, dispatch }}>
      <LoginRouter />;
    </AuthContext.Provider>
  );
};

export default App;
