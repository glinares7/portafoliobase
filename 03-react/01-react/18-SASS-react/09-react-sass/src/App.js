import React from "react";
import Button from "./components/Button";

const App = () => {
  // <button className="btn btn-primary">+1</button>
  return (
    <div>
      <h1 className="title">Hola desde - SASS</h1>
      <hr />
      <ul>
        <li>
          <a href="#">Google</a>
        </li>
        <li>
          <a href="#">Facebook</a>
        </li>
        <li>
          <a href="#">Instagram</a>
        </li>
      </ul>
      <hr />
      <h2>Counter:0</h2>
      <Button type="btn btn-primary" text="+1" />
      <Button type="btn btn-secondary" text="-1" />
    </div>
  );
};

export default App;
