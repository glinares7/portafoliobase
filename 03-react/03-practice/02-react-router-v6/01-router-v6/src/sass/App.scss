$valor: red;
$fondo: skyblue;

.app__body {
  position: relative;
  width: 100%;
  height: 100%;
  padding-bottom: 54px;
  min-height: calc(100vh - 54px);
  background-color: yellow;
}
