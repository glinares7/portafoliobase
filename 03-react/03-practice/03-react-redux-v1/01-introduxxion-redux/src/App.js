import Main from "./components/Main";

// import { Provider } from "react-redux";

function App() {
  // store.dispatch({
  //   type: "DECREMENT",
  // });

  return <Main />;
}

export default App;
