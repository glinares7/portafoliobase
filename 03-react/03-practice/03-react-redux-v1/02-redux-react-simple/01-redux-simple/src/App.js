import { useDispatch, useSelector } from "react-redux";
import { Link } from "react-router-dom";
import { store } from "./index";
function App() {
  const dispatch = useDispatch();
  const myCounter = useSelector((state) => state.outCounter.count);

  //* estado actual
  console.log("*******mi estado es ******", myCounter);

  const unsubscribe = store.subscribe(() => {
    // vemos el nuevo store
    console.log("suscribe watch", store.getState());
    // nos desuscribimos
    unsubscribe();
  });
  return (
    <div
      style={{
        paddingLeft: "20px",
        display: "flex",
        width: "100%",
        minHeight: "90vh",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      <div style={{ width: "50%" }}>
        <h1>Simple redux count</h1>
        <h1>contador + - : {myCounter}</h1>
        <button onClick={() => dispatch({ type: "INCREMENT" })}>+</button>
        <button onClick={() => dispatch({ type: "DECREMENT" })}>-</button>
        <button
          onClick={() => dispatch({ type: "RESET", payload: { count: 0 } })}
        >
          RESET
        </button>
        <br />
        <br />
        <Link to="somes">totales</Link>
      </div>
    </div>
  );
}

export default App;
