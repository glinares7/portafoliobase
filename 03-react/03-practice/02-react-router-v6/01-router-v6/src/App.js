import "./sass/App.scss";

import { UseContext } from "./contexts/AuthContext";
// import { themes } from "./helpers/Pictures";
import { useState } from "react";

import AppRouter from "./routes/AppRouter";

const App = () => {
  const [appcontext, setAppcontext] = useState();
  const [sliderc, setSliderc] = useState();

  const [log, setLog] = useState(false);

  const [user, setUser] = useState(null);

  const fakeAuth = {
    isAuthenticated: false,
    signin(cb) {
      fakeAuth.isAuthenticated = true;
      setTimeout(cb, 100); // fake async
    },
    signout(cb) {
      fakeAuth.isAuthenticated = false;
      setTimeout(cb, 100);
    },
  };

  const signin = (cb) => {
    return fakeAuth.signin(() => {
      setUser("user");
      cb();
    });
  };

  const signout = (cb) => {
    return fakeAuth.signout(() => {
      setUser(null);
      cb();
    });
  };

  return (
    <UseContext.Provider
      value={{
        appcontext,
        setAppcontext,
        sliderc,
        setSliderc,
        log,
        setLog,
        user,
        setUser,
        signin,
        signout,
      }}
    >
      <AppRouter />
    </UseContext.Provider>
  );
};

export default App;
