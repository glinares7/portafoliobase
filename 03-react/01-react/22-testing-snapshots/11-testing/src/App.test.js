import React from "react";
import { render, screen } from "@testing-library/react";
import App from "./App";

// const linkElement = screen.getByText(/learn react/i);
test("renders learn react link", () => {
  render(<App />);
  expect(screen.getByRole("app")).toMatchSnapshot();
  // const linkElement = screen.getByText("Counter: 0");
  // expect(linkElement).toBeInTheDocument();
});
