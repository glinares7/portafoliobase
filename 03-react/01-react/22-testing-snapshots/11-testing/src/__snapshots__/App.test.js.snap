// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders learn react link 1`] = `
<div
  role="app"
>
  <h2
    role="counter"
  >
    Counter: 
    0
  </h2>
  <button
    aria-label="aumentar"
  >
    1
  </button>
  <button
    aria-label="reset"
  >
    Reset
  </button>
  <button
    aria-label="disminuir"
  >
    -1
  </button>
</div>
`;
